build_tools/circle/build_doc.sh

```diff
@@ -58,6 +58,25 @@ get_build_type() {
         return
     fi
     changed_examples=$(echo "$filenames" | grep -E "^examples/(.*/)*plot_")
+    if [ -n "$filenames" ]
+    then
+        for af in ${filenames[@]}
```

Comment:
For this section how do you feel about:

```bash
if [ -n "$filenames"] then
    examples_in_rst=$(echo "$filenames" | \
					  grep -E "rst$" | \
 					  xargs grep -shE "(figure|image)::" | \
					  grep auto_examples | \
					  awk -F "../auto_examples/" '{print $NF}' | \
					  sed 's/images\/sphx_glr_//' | \
					  sed -e 's/_[[:digit:]][[:digit:]][[:digit:]].png/.py/' | \
					  uniq | \
					  sed -e 's/^/examples\//')
	changed_examples="${changed_examples}\n${examples_in_rst}"
fi

```

Reply:
... well that's awesome... and clearly you had a lot of fun in building it! ;)
Is that a cryptic way to ask me to put your code in this PR? :)
Thanks for the `[:digit:]` thing: I looked for it for an entire afternoon without luck... not the right keywords on google search clearly...